test/server_integration/http/ssl_redirect/config.js

```diff
@@ -38,20 +37,18 @@ export default async function ({ readConfigFile }) {
   const httpConfig = await readConfigFile(require.resolve('../../config'));
   const certificateAuthorities = [readFileSync(CA_CERT_PATH)];
 
-  const redirectPort = httpConfig.get('servers.opensearchDashboards.port') + 1234;
+  const opensearchDashboardsUrl = new URL('', httpConfig.get('servers.opensearchDashboards'));
+  const redirectPort = opensearchDashboardsUrl.port + 1234;
+  opensearchDashboardsUrl.port = redirectPort;
+  opensearchDashboardsUrl.protocol = 'http';
 
   return {
     testFiles: [require.resolve('./')],
     services: {
       ...httpConfig.get('services'),
       supertest: createOpenSearchDashboardsSupertestProvider({
         certificateAuthorities,
-        opensearchDashboardsUrl: Url.format({
-          ...httpConfig.get('servers.opensearchDashboards'),
-          port: redirectPort,
-          // test with non ssl protocol
-          protocol: 'http',
-        }),
```

Comment:
`createOpenSearchDashboardsSupertestProvider` is called in a bunch of other files too; shouldn't those files be changed too?

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/d7004dc5b0392477fdd54ac66b29d231975a173b/test/server_integration/http/ssl_with_p12_intermediate/config.js#L49-L53

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/d7004dc5b0392477fdd54ac66b29d231975a173b/test/server_integration/http/ssl_with_p12/config.js#L49-L53
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/d7004dc5b0392477fdd54ac66b29d231975a173b/test/server_integration/http/ssl/config.js#L49-L53

Reply:
You are perfectly right. Seems I missed some case for `servers.opensearchDashboards`. I will update this part.